### ISSUE TITLE
unexported-return: unexported interfaces should not be allowed

### ISSUE THREAD

**Issue by tie** [mgechev/revive#1457]
Currently, `unexported-return` rule allows returning unexported interface types. E.g.

```go
type unexportedInterface interface {
        unexportedMethod()
}

func New() unexportedInterface {
        return nil
}
```

As per discussion in https://github.com/mgechev/revive/pull/1456#issuecomment-3146371336, this is unfriendly to the user of the package. The example above is also “annoying to use“ because the interface type includes an unexported method, so it cannot be assigned to a value that is a subset of the original interface in different package. E.g. https://go.dev/play/p/ohypfiXhADR

**Expected behavior**

The `unexported-return` rule should not allow named unexported return types (except when defined as an exported type alias).
